src/main/java/com/amplitude/Amplitude.java

```diff
@@ -24,6 +25,8 @@ public class Amplitude {
   private Plan plan;
   private IngestionMetadata ingestionMetadata;
   private long flushTimeout;
+  private ExecutorService retryThreadPool;
```

Comment:
@kevink-sq - trying to figure out why we are setting this here. Isn't it sufficient that the setters set the value directly in the HttpTransport?

Reply:
The only way we can create an instance of amplitude is by calling `getInstance` and when that is called, this value will always be null.

Reply:
@izaaz - You are right. I thought we could set the thread pools first before instantiation. Similar implementation was also done for `flushTimeout`. I'll clean up some of the code.

Reply:
bump @izaaz

Reply:
@kevink-sq - I see that you're cleaning up the code around this and thank you for that. For this PR, can we limit the change to just the change you want? I'll create a separate internal ticket to clean up this code. This SDK has not been changed in a while and I'd limit the changes here just to make sure nothing else goes wrong (though your changes look right).

Reply:
@izaaz scoped the PR to only my change